exec/java-exec/src/main/java/org/apache/drill/exec/util/FileSystemUtil.java

```diff
@@ -55,7 +53,28 @@ public boolean accept(Path path) {
    */
   public static List<FileStatus> listDirectories(final FileSystem fs, Path path, boolean recursive, PathFilter... filters) throws IOException {
     List<FileStatus> statuses = new ArrayList<>();
-    listDirectories(fs, path, recursive, statuses, mergeFilters(filters));
+    listDirectories(fs, path, recursive, false, statuses, mergeFilters(filters));
+    return statuses;
+  }
+
+  /**
+   * Returns statuses of all directories present in given path applying custom filters if present.
+   * Will also include nested directories if recursive flag is set to true.
+   * Will ignore all exceptions during listing if any.
+   *
+   * @param fs current file system
+   * @param path path to directory
+   * @param recursive true if nested directories should be included
+   * @param filters list of custom filters (optional)
+   * @return list of matching directory statuses
+   */
+  public static List<FileStatus> listDirectoriesSafe(final FileSystem fs, Path path, boolean recursive, PathFilter... filters) {
+    List<FileStatus> statuses = new ArrayList<>();
+    try {
+      listDirectories(fs, path, recursive, false, statuses, mergeFilters(filters));
```

Comment:
Looks like `suppressExceptions` should be `true` here.

Reply:
Good catch, thanks!